File: java/src/main/java/de/gdata/vaas/exceptions/VaasClientException.java
package de.gdata.vaas.exceptions;

/**
 * The request is malformed or cannot be completed.
 * Recommended actions:
 * * Don't repeat the request.
 * * Log.
 * * Analyze the error.
 */
public class VaasClientException extends Exception {
    public VaasClientException(String message, Exception cause) {
        super(message != null ? message : "Client error", cause);
    }

    public VaasClientException(String message) {
        this(message, null);
    }

    public VaasClientException() {
        this(null, null);
    }
}

File: dotnet/Vaas/src/Vaas/Exceptions/VaasClientException.cs
using System;

namespace Vaas.Exceptions;

/// <summary>The request is malformed or cannot be completed.</summary>
/// <remarks>
/// Recommended actions:
/// <ul>
///   <li>Don't repeat the request.</li>
///   <li>Log.</li>
///   <li>Analyze the error</li>
/// </ul>
/// </remarks>
public class VaasClientException : Exception
{
    public VaasClientException(string? message)
        : base(message) { }

    public VaasClientException(string? message, Exception? innerException)
        : base(message, innerException) { }
}
